### PR TITLE
add plot of confusion matrix

### PR DESCRIPTION
**Pull request by xinyuejohn**
Summary

> Plot confusion matrix after training.
> Save confusion matrix 

Implement Details

> Convert computed confusion matrix to pandas.DataFrame and plot it
> Save confusion matrix as .csv in zoutput/aggrsts
> Use plot function from rich package to get better formatting in the terminal.

Example

Using the following code:
`python -W ignore main_out.py --te_d=caltech --task=mini_vlcs --aname=diva --npath=examples/nets/resnet.py --npath_dom=examples/nets/resnet.py --gamma_y=7e5 --gamma_d=1e5 --epos=2`

Screenshot 2023-02-02 at 22 21 59

### CODE REVIEW

**Comment by xinyuejohn** [comment]
@smilesun As you can see from the above image, for each run, there will be one confusion matrix at the end. 

In which way do you suggest saving the confusion matrix? Because for the other results (precision, recall, ... ), every time an extra line will be appended to the .csv file. But for confusion matrix, if is is saved in a .csv file, it is hard to to so.

**Comment by smilesun** [comment]
@xinyuejohn , this is a feature proposed by @aliechoes , maybe he has some ideas on how the feature should be?

**Comment by aliechoes** [comment]
Thanks for asking. I would like to propose that we save the confusion matrix as a plot using sklearn.metrics.ConfusionMatrixDisplay

**Comment by smilesun** [comment]
> @smilesun Also, do you think it is necessary to add the additional dependency of rich package and its output effect?
> 
> If you like it and want to add it, could you please fix the CI test? Because it doesn't use the current .toml file to create the test environment.
@xinyuejohn 
I hope my two commits will resolve this dependences, also I think what @aliechoes suggested is really good, it will be cool to save a pdf or svg to the zoutput folder after a run.

**Review by smilesun — CHANGES_REQUESTED** [review]
there is codacy issue with trailing white spaces somewhere

**Review by smilesun — CHANGES_REQUESTED** [review]
please see the individual line wise reviews, especially we want to avoid hard coded strings, welcome for discussions

File: domainlab/compos/exp/exp_utils.py [hunk 1]
```diff
@@ -161,3 +163,18 @@ def to_file(self, str_line):
...
+    def confmat_to_file(self, confmat, confmat_filename):
+        """Save confusion matrix as a figure
+
+        Args:
+            confmat: confusion matrix.
+        """
+        disp = ConfusionMatrixDisplay(confmat)
+        disp = disp.plot(cmap="gray")
+        file_path = self.get_fpath()
+        file_path = file_path.removesuffix(".csv")
```
[the remark this answers is not in the document]
Comment (smilesun) [reply domainlab/compos/exp/exp_utils.py:177]: i added fixedme

**Comment by smilesun** [comment]
i added some fixeme

**Pull request merged**